lib/msal-angular/docs/initialization-2.x.md

```diff
@@ -0,0 +1,147 @@
+# Initialization of MSAL Angular
+
+Before using MSAL.js, [register an application in Azure AD](https://docs.microsoft.com/azure/active-directory/develop/quickstart-register-app) to get your `clientId`.
+
+In this document:
+- [Initialization of MSAL](#initialization-of-msal-angular)
+    - [Include and initialize the MSAL module in your app module](#include-and-initialize-the-msal-module-in-your-app-module)
+    - [Secure the routes in your application](#secure-the-routes-in-your-application)
+    - [Get tokens for Web API calls](#get-tokens-for-web-api-calls)
+    - [Subscribe to event callbacks](#subscribe-to-event-callbacks)
+- [Next Steps](#next-steps)
+
+
+
+## Include and initialize the MSAL module in your app module
+
+Import MsalModule into app.module.ts. To initialize MSAL module you are required to pass the clientId of your application which you can get from the application registration.
+
+```js
+@NgModule({
+    imports: [
+        MsalModule.forRoot({
+            auth: {
+                clientId: "Your client ID"
+            }
+        })
+    ]
+})
+export class AppModule {}
+```
+
+## Secure the routes in your application
+
+You can add authentication to secure specific routes in your application by just adding `canActivate : [MsalGuard]` to your route definition. It can be added at the parent or child routes.
+
+```js
+{
+    path: 'product',
+    component: ProductComponent,
+    canActivate: [MsalGuard],
+    children: [
+        {
+            path: 'detail/:id',
+            component: ProductDetailComponent
+        }
+    ]
+}, {
+    path: 'myProfile',
+    component: MsGraphComponent,
+    canActivate: [MsalGuard]
+},
+```
+
+When a user visits these routes, the library will prompt the user to authenticate.
+
+## Get tokens for Web API calls
+
+MSAL Angular allows you to add an Http interceptor (`MsalInterceptor`) in your `app.module.ts` as follows. MsalInterceptor will obtain tokens and add them to all your Http requests in API calls based on the `protectedResourceMap`.
+
+```js
+@NgModule({
+    imports: [
+        MsalModule.forRoot({ // MSAL Configuration
+            auth: {
+                clientId: "Your client ID"
+            }
+        }, {
+            interactionType: InteractionType.Popup, // MSAL Guard Configuration
+            authRequest: PopupRequest
+        }, {
+            protectedResourceMap: new Map([ // MSAL Interceptor Configuration
+                ['https://graph.microsoft.com/v1.0/me', ['user.read']],
+                ['https://api.myapplication.com/users/*', ['customscope.read']]
```

Comment:
We should show how setting a scope value of `null` or a resource will prevent it from getting tokens (a la unprotectedResources)

Reply:
Can you double check this was done correctly?